cli/pub_command.go

```diff
@@ -119,7 +120,23 @@ func (c *pubCmd) doReq(nc *nats.Conn, progress *uiprogress.Bar) error {
 			return err
 		}
 
-		m, err := nc.RequestMsg(msg, opts.Timeout)
+		var m *nats.Msg
+		if c.replyTo == "" {
+			m, err = nc.RequestMsg(msg, opts.Timeout)
+			if err != nil {
+				return err
+			}
+		} else {
+			sub, e := nc.SubscribeSync(c.replyTo)
+			if e != nil {
+				return e
+			}
+			if e := nc.PublishMsg(msg); e != nil {
+				return e
+			}
+			m, err = sub.NextMsg(opts.Timeout)
+		}
+
 		if err != nil {
```

Comment:
move the error check up so it checks after the err is set, easy mistake to make later to drop this or insert stuff in between

Reply:
fixed